cpp/cmake/Modules/ConfigureArrow.cmake

```diff
@@ -6,19 +6,19 @@ set(ARROW_CMAKE_ARGS " -DARROW_WITH_LZ4=OFF"
                      " -DARROW_WITH_SNAPPY=OFF"
                      " -DARROW_WITH_ZLIB=OFF"
                      " -DARROW_BUILD_STATIC=ON"
-                     " -DARROW_BUILD_SHARED=OFF"
+                     " -DARROW_BUILD_SHARED=ON"
                      " -DARROW_BOOST_USE_SHARED=OFF"
                      " -DARROW_BUILD_TESTS=OFF"
                      " -DARROW_TEST_LINKAGE=OFF"
                      " -DARROW_TEST_MEMCHECK=OFF"
                      " -DARROW_BUILD_BENCHMARKS=OFF"
                      " -DARROW_IPC=ON"
                      " -DARROW_FLIGHT=OFF"
-                     " -DARROW_COMPUTE=OFF"
+                     " -DARROW_COMPUTE=ON"
                      " -DARROW_CUDA=ON"
                      " -DARROW_JEMALLOC=OFF"
                      " -DARROW_BOOST_VENDORED=OFF"
-                     " -DARROW_PYTHON=OFF"
+                     " -DARROW_PYTHON=ON"
```

Comment:
The arrow being built here isn't installed on the system so we only build the minimal subset that we need for static linking. Why are we turning these on?

Reply:
When not using conda to install pyarrow those python libs were needed during that installation. You are right that doesn't have to necessarily happen here but somewhere else in a build script it seems it would need to be built and figured since we were 95% there here to just use this build for that. What do you think we should do there?

Reply:
So with 0.17.1 Arrow builds with the CUDA extensions on the way in conda-forge this code will be ripped out soon. Given there's not a CUDA enabled pip package, the only choice for end users will be to build Arrow / PyArrow from source and install themselves, then point to the install via typical `ARROW_ROOT` environment variable.

Reply:
ok so since the end goal is a build on Jetson devices sounds like the best option is to just build Arrow/PyArrow separately in the Jetson build script would you agree?

Also any insight on when that 0.17.1 change will be happening?